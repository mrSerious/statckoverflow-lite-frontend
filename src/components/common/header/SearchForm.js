import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { searchRequest } from '../../../actions/searchAction';

export class SearchForm extends Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.state = { value: '' };
  }

  handleChange(event) {
    this.setState({ value: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    const { value } = this.state;
    const { search, history } = this.props;
    history.push('/results');
    search(value);
  }

  render() {
    const { value } = this.state;
    return (
      <form onSubmit={this.handleSubmit} className="site-navigation-search">
        <input
          type="text"
          placeholder="Search for..."
          name="search"
          value={value}
          onChange={this.handleChange}
        />
      </form>
    );
  }
}

SearchForm.propTypes = {
  search: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};


const mapStateToProps = state => ({
  loading: state.search.loading,
});

export const mapDispatchToProps = dispatch => ({
  search: (searchTerm) => {
    dispatch(searchRequest(searchTerm));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchForm);
