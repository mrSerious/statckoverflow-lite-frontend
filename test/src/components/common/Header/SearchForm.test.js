import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from '../../../../../src/components/common/header/SearchForm';

describe('Header tests', () => {
  const wrapper = shallow(<SearchForm />);

  it('should render the search form without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
