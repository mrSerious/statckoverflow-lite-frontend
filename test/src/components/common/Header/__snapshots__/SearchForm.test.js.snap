// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search form tests should render the search form without crashing 1`] = `
<form
  className="site-navigation-search"
  onSubmit={[Function]}
>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search for..."
    type="text"
    value=""
  />
</form>
`;
